ledger-tool/src/main.rs

```diff
@@ -1621,7 +1637,96 @@ fn main() {
                         },
                     );
 
-                    let process_options = parse_process_options(&ledger_path, arg_matches);
+                    let mut process_options = parse_process_options(&ledger_path, arg_matches);
+
+                    let include_bank = match arg_matches.value_of("verify_slots_details").unwrap() {
+                        "none" => false,
+                        "bank" => true,
+                        _ => unreachable!(),
+                    };
+
+                    // if --verify-slots=file is not specified: do nothing
+                    // if --verify-slots=file is specified and file exists: check slot hashes against file
+                    // if --verify-slots=file is specified and file does not exist: write file with or with or without
+                    //    bank details depending on whether --verify_slots_details=bank is specified or not.
+                    let (slot_callback, record_slots_file, recorded_slots) = if let Some(filename) =
+                        arg_matches.value_of_os("verify_slots")
+                    {
+                        let filename = Path::new(filename);
+
+                        if filename.exists() {
+                            let file = File::open(filename).unwrap_or_else(|err| {
+                                eprintln!("Unable to read file: {}: {err:#}", filename.display());
+                                exit(1);
+                            });
+
+                            let reader = std::io::BufReader::new(file);
+
+                            let details: bank_hash_details::BankHashDetails =
+                                serde_json::from_reader(reader).unwrap_or_else(|err| {
+                                    eprintln!("Error loading slots file: {err:#}");
+                                    exit(1);
+                                });
```

Comment:
Does this run without error as-is? The other fields in `BankHashDetails` are not optional (parent_bank_hash, accounts_delta_hash, etc), so I would expect this deserialize to fail if they're not present.

And similar to other comment, I'd like to keep as much impl as possible over in `bank_hash_details.rs`. Obviously the slot_callback is unique to this file, but the I/O could seemingly be shifted over there.

Reply:
Those fields just have default values if we do hash-only/

```
    {
      "slot": 102,
      "bank_hash": "4JH7J2TjyMwWBcx6rAcYnSBhR76mWxBYpmnMVWruDmQs",
      "parent_bank_hash": "",
      "accounts_delta_hash": "",
      "signature_count": 0,
      "last_blockhash": "",
      "accounts": []
    },
```

I've added a bunch of serde options so these empty fields don't get serialized. It clutters the `BankHashSlotDetails` definition, let me know what you think.

Reply:
> Those fields just have default values if we do hash-only/

Got it yeah, I confused myself there. You're only populating the slot / bank_hash fields, but every field was getting serialized. So, when you went to deserialize in this snippet, all the fields present.

> I've added a bunch of serde options so these empty fields don't get serialized. It clutters the BankHashSlotDetails definition, let me know what you think

Agreed that it adds some clutter to the definition, but I like it. Will make viewing of the bank hashes only much more pleasant. Do the tags you added work for both deserialization too (ie if the fields are absent, it will populate as default) ?

Reply:
`--verify-slots` works fine without `--verify-slots-config=accounts`, I've been testing it as a I go along (I'm debugging consensus failures rn using this)